src/server.rs

```diff
@@ -289,7 +319,7 @@ impl LanguageServer for LspServer {
             }),
         })
     }
-    async fn shutdown(&self) -> Result<()> {
+    async fn shutdown(&self) -> RpcResult<()> {
```

Comment:
Can we back this alias out? It makes the patch bigger than it needs to be, and I don't see a need to not shadow `std::Result` here (I suspect it's here because some code got moved in and then moved back out that needed it).